dbms/src/DataStreams/UnionBlockInputStream.h

```diff
@@ -255,6 +255,16 @@ class UnionBlockInputStream final : public IProfilingBlockInputStream
             children[i]->readSuffix();
     }
 
+    uint64_t collectCPUTimeNsImpl(bool /*is_root*/) override
+    {
+        uint64_t cpu_time_ns = 0;
```

Comment:
Why ignore `is_root` here?

Reply:
`is_root` is a bad name...
I had renamed to `is_thread_runner` and add some comments here.
```
    /** Estimate the cpu time nanoseconds used by block input stream dag.
      * In this method, streams are divided into two categories:
      * - thread-runner: Called directly by a thread
      * - non-thread-runner: Called by a thread-runner
      * Here we should count the execution time of each thread-runner.
      * Note: Because more threads than vcore, and blocking relationships between streams,
      * the result may not be 100% identical to the actual cpu time nanoseconds.
      */

        // UnionBlockInputStream isn't a thread-runner, so is_thread_runner is ignored here.
        // But each of UnionBlockInputStream's children is a thread-runner.
```

Reply:
If `is_root` is equivalent to `is_thread_runner`? if true, we should add comment about it.

Reply:
> If `is_root` is equivalent to `is_thread_runner`? if true, we should add comment about it.

yes, comment `// The first stream of stream dag is thread-runner.` added.